### PR TITLE
Fix 02932_kill_query_sleep flakiness

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Cc: @vitlibar

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit f7fa48d5668a1c84a2b26833e3c36f1b35c8fa5b with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57849/f7fa48d5668a1c84a2b26833e3c36f1b35c8fa5b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7206068289">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57849/f7fa48d5668a1c84a2b26833e3c36f1b35c8fa5b/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/queries/0_stateless/02932_kill_query_sleep.sh [hunk 1]
```diff
@@ -8,6 +8,7 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 function wait_query_started()
 {
     local query_id="$1"
+    $CLICKHOUSE_CLIENT --query "SYSTEM FLUSH LOGS"
```
Comment (vitlibar) [inline tests/queries/0_stateless/02932_kill_query_sleep.sh:11]: Why was it flaky?

Comment (azat) [reply tests/queries/0_stateless/02932_kill_query_sleep.sh:11]: A line below it uses `system.query_log` which can be not available yet.

Comment (azat) [reply tests/queries/0_stateless/02932_kill_query_sleep.sh:11]: https://github.com/ClickHouse/ClickHouse/pull/55271#issuecomment-1854820934

https://s3.amazonaws.com/clickhouse-test-reports/56563/8edabdfd61a7c7c37f57379fc2a40a3d25e4e7bc/fast_test.html

Comment (vitlibar) [reply tests/queries/0_stateless/02932_kill_query_sleep.sh:11]: So we don't create `system.query_log` on SELECT, only on flushing? This is strange...

**Review by vitlibar — APPROVED** [review]

**Pull request merged**